### PR TITLE
Nada/FEQ-2481/p2p dr

### PR DESCRIPTION
**Pull request by nada-deriv**
This pr adds the vercel job to the production and staging github workflow for p2p.

### CODE REVIEW

File: .github/workflows/build-and-deploy-staging.yml [hunk 1]
```diff
@@ -58,3 +58,17 @@ jobs:
...
+        steps:
+            - name: Upload to vercel
+              uses: 'deriv-com/shared-actions/.github/actions/vercel_DR_publish@master'
+              with:
+                VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+                VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+                VERCEL_TOKEN: ${{ secrets.VERCEL_API_TOKEN }}
+                ENVIRONMENT: Preview
+                VERCEL_SCOPE: deriv
+                ALIAS_DOMAIN_URL: 'staging-p2p-dr.deriv.com'
```
Comment (ali-hosseini-deriv) [inline .github/workflows/build-and-deploy-staging.yml:74]:
```suggestion
                ALIAS_DOMAIN_URL: 'staging-p2p-dr.binary.sx'
```

File: .github/workflows/build-and-deploy-prod.yml [hunk 2]
```diff
@@ -57,14 +57,42 @@ jobs:
...
+        steps:
+            - name: Upload to vercel
+              uses: 'deriv-com/shared-actions/.github/actions/vercel_DR_publish@master'
+              with:
+                VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+                VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+                VERCEL_TOKEN: ${{ secrets.VERCEL_API_TOKEN }}
+                ENVIRONMENT: Production
+                VERCEL_SCOPE: deriv
+                ALIAS_DOMAIN_URL: 'p2p-dr.deriv.com'
```
Comment (ali-hosseini-deriv) [inline .github/workflows/build-and-deploy-prod.yml:72]:
```suggestion
                ALIAS_DOMAIN_URL: 'p2p-dr.binary.sx'
```

File: .github/workflows/build-and-deploy-prod.yml [hunk 3]
```diff
@@ -57,14 +57,42 @@ jobs:
...
-          uses: "deriv-com/shared-actions/.github/actions/send_slack_notification@master"
-          with:
-            SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-            MESSAGE: "Production Release succeeded for p2p.deriv.com with version ${{ github.head_ref }}"
+    publish_to_vercel:
+        runs-on: ubuntu-latest
+        environment: Production
+        steps:
+            - name: Upload to vercel
+              uses: 'deriv-com/shared-actions/.github/actions/vercel_DR_publish@master'
```
Comment (ali-hosseini-deriv) [inline .github/workflows/build-and-deploy-prod.yml:65]:
```suggestion
              uses: 'deriv-com/shared-actions/.github/actions/vercel_DR_publish@master'
              id: dr_action
```

File: .github/workflows/build-and-deploy-prod.yml [hunk 4]
```diff
@@ -57,14 +57,42 @@ jobs:
...
+              with:
+                VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+                VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+                VERCEL_TOKEN: ${{ secrets.VERCEL_API_TOKEN }}
+                ENVIRONMENT: Production
+                VERCEL_SCOPE: deriv
+                ALIAS_DOMAIN_URL: 'p2p-dr.deriv.com'
+            - name: Send Slack Notification on Vercel Publish failure
+              uses: "deriv-com/shared-actions/.github/actions/send_slack_notification@master"
```
Comment (ali-hosseini-deriv) [inline .github/workflows/build-and-deploy-prod.yml:74]:
```suggestion
              uses: "deriv-com/shared-actions/.github/actions/send_slack_notification@master"
              if: steps.dr_action.conclusion == 'failure'
```

Comment (ali-hosseini-deriv) [reply .github/workflows/build-and-deploy-prod.yml:74]: This is to make sure that only if the the previous job fails we are sending the slack message.

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]

**Review by ali-hosseini-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**